### DISCUSSION TITLE
Emulating Tock on software

### DISCUSSION THREAD

**Discussion by AndrewImwalle** [tock/tock#3065] [Q&A] [answered]
I am trying to build TockOS in a pure software environment for x86. Is there documentation or has anyone explored this path before

**Comment by jrvanwhy** [comment]
The Ti50 project has a "host emulation" mode, in which Tock is compiled with an architecture and "chip" implementation that allow it to run in a standard Linux process. Apps can run as well, each as their own Linux process, and the host emulation environment uses IPC mechanisms to pass syscalls between the apps and kernel.

Is that what you're referring to? If so, it's already been proven possible. I don't think the implementation is public, however. If that's not what you're referring to, then please clarify.

**Comment by alexandruradovici** [comment]
If you need emulating Tock on something that runs on an x86 machine, you can use [renode](https://renode.io/). It can emulate an STM32F4 Discovery board, with limited peripherals (gpio and uart). We have a github [template](https://github.com/Wyliodrin/rust-tock-project) that you can start from.

**Comment by hudson-ayers** [comment, marked as answer]
There was a start on native x86 support here: https://github.com/tock/tock/pull/2111 if you are interested in picking up that work
